### PR TITLE
ci: refactor workflows

### PR DESCRIPTION
**Pull request by fpetkovski**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -143,9 +143,9 @@ BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 
 .PHONY: bundle
 bundle: $(KUSTOMIZE) $(OPERATOR_SDK) generate
-	cd deploy/operator && \
-		$(KUSTOMIZE) edit set image monitoring-stack-operator=$(OPERATOR_IMG) && \
-		$(KUSTOMIZE) edit set label app.kubernetes.io/version:$(VERSION)
+	cd deploy/olm && \
```
Comment (fpetkovski) [inline Makefile:146]: deploy/operator are agnostic manifests used for local deployment as well.

File: deploy/operator/kustomization.yaml [hunk 2]
```diff
@@ -5,12 +5,3 @@ resources:
...
 - serviceaccount.yaml
 
-apiVersion: kustomize.config.k8s.io/v1beta1
-kind: Kustomization
-images:
-- name: monitoring-stack-operator
-  newName: quay.io/sthaha/monitoring-stack-operator
-  newTag: 0.0.1
-
-commonLabels:
```
Comment (fpetkovski) [inline deploy/operator/kustomization.yaml:15]: commonLabels also sets deployment selector which are immutable and should not be changed.

File: .github/workflows/olm-candidate.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  release:
+    runs-on: ubuntu-latest
+    environment: quay
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set version
+        id: version
+        run: |
+          version="$(cat VERSION)-rc"
```
Comment (sthaha) [inline .github/workflows/olm-candidate.yaml:19]: would this mean, the previous candidate will be overwritten by this version?

File: deploy/operator/kustomization.yaml [hunk 4]
```diff
@@ -5,12 +5,3 @@ resources:
 - role.yaml
 - serviceaccount.yaml
 
-apiVersion: kustomize.config.k8s.io/v1beta1
-kind: Kustomization
-images:
-- name: monitoring-stack-operator
```
Comment (fpetkovski) [inline deploy/operator/kustomization.yaml:11]: Moved to deploy/olm/kustomization.yaml

**Pull request merged**